test/e2e/install_test.go

```diff
@@ -99,10 +109,137 @@ var _ = Describe("Operator Install", func() {
+		Expect(err).To(Not(HaveOccurred()))
+		err = ioutil.WriteFile(artifactPath+operator.Name+"-operator.yaml", operatorYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	// get all catalogsources save them to the artifact path.
+	catalogsources := catalogd.CatalogSourceList{}
+	err = c.List(ctx, &catalogsources, client.InNamespace(""))
+	Expect(err).To(Not(HaveOccurred()))
+	for _, catalogsource := range catalogsources.Items {
+		// Save catalogsource to artifact path
+		catalogsourceYaml, err := yaml.Marshal(catalogsource)
+		Expect(err).To(Not(HaveOccurred()))
+		err = ioutil.WriteFile(artifactPath+catalogsource.Name+"-catalogsource.yaml", catalogsourceYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	// Get all Bundles in the namespace and save them to the artifact path.
+	bundles := rukpakv1alpha1.BundleList{}
+	err = c.List(ctx, &bundles, client.InNamespace(""))
+	Expect(err).To(Not(HaveOccurred()))
+	for _, bundle := range bundles.Items {
+		// Save bundle to artifact path
+		bundleYaml, err := yaml.Marshal(bundle)
+		Expect(err).To(Not(HaveOccurred()))
+		err = ioutil.WriteFile(artifactPath+bundle.Name+"-bundle.yaml", bundleYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	// Get all BundleDeployments in the namespace and save them to the artifact path.
+	bundleDeployments := rukpakv1alpha1.BundleDeploymentList{}
+	err = c.List(ctx, &bundleDeployments, client.InNamespace(""))
+	Expect(err).To(Not(HaveOccurred()))
+	for _, bundleDeployment := range bundleDeployments.Items {
+		// Save bundleDeployment to artifact path
+		bundleDeploymentYaml, err := yaml.Marshal(bundleDeployment)
+		Expect(err).To(Not(HaveOccurred()))
+		err = ioutil.WriteFile(artifactPath+bundleDeployment.Name+"-bundleDeployment.yaml", bundleDeploymentYaml, 0644)
+		Expect(err).To(Not(HaveOccurred()))
+	}
+
+	for _, namespace := range namespaces.Items {
+		// let's ignore kube-* namespaces.
+		if strings.Contains(namespace.Name, "kube-") {
+			continue
+		}
+
+		namespaceArtifactPath := artifactPath + namespace.Name + "/"
+		err = os.Mkdir(namespaceArtifactPath, 0755)
+		Expect(err).To(Not(HaveOccurred()))
+
+		// get all deployments in the namespace and save them to the artifact path.
+		deployments, err := kubeClient.AppsV1().Deployments(namespace.Name).List(ctx, metav1.ListOptions{})
+		Expect(err).To(Not(HaveOccurred()))
+		for _, deployment := range deployments.Items {
+			// Save deployment to artifact path
+			deploymentYaml, err := yaml.Marshal(deployment)
+			Expect(err).To(Not(HaveOccurred()))
+			err = ioutil.WriteFile(namespaceArtifactPath+deployment.Name+"-deployment.yaml", deploymentYaml, 0644)
+			Expect(err).To(Not(HaveOccurred()))
+		}
+
+		// Get logs from all pods in all namespaces
+		pods, err := kubeClient.CoreV1().Pods(namespace.Name).List(ctx, metav1.ListOptions{})
+		Expect(err).To(Not(HaveOccurred()))
+		for _, pod := range pods.Items {
+			if pod.Status.Phase != v1.PodRunning {
```

Comment:
You could also try to get logs from completed/terminated pods?

Reply:
fixed